packages/headless/src/features/folding/folding-actions.ts

```diff
@@ -1,14 +1,35 @@
 import {NumberValue, SchemaDefinition, StringValue} from '@coveo/bueno';
-import {createAction} from '@reduxjs/toolkit';
+import {createAction, createAsyncThunk} from '@reduxjs/toolkit';
+import {
+  AsyncThunkSearchOptions,
+  isErrorResponse,
+} from '../../api/search/search-api-client';
+import {SearchAPIErrorWithStatusCode} from '../../api/search/search-api-error-response';
+import {Result} from '../../api/search/search/result';
+import {ConfigurationSection, FoldingSection} from '../../state/state-sections';
 import {validatePayload} from '../../utils/validate-payload';
 
+export const collectionMoreResultsAvailableBuffer = 1;
```

Comment:
It's typically good to store magic numbers in well named variables, but in this case I see it as a leaky abstraction. We should try to hide this detail.

For example, when the folding is registered, can we add 1? Or should there be a new field in the folding slice called `filterFieldRange`?

The goal should be for the [search request code](https://github.com/coveo/ui-kit/pull/791/files#diff-693a41de1ad1f40792efcdd3272198d1336232eab9da456f2a35bc893cee6038R330) to not need to perform an addition. This knowledge should be abstracted away,

Reply:
We could, but I'm worried that it would become confusing in the reducer why we remove one folded result, and harder to track which parts of the code handle the show more buffer.

Reply:
We could keep the `1` stored in a variable, but the idea is to not have the variable be exported outside of the folding reducer. The rest of the code-base would ideally not need to know about this detail, need to import the variable and perform an arithmetic operation with it,